python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,423 @@
+
+    @property
+    def id(self):
+        """The id of the item.
+
+        This id is unique to the item and is persistent across serialization/
+        deserialization.
+        """
+        return self._id
+
+    @property
+    def local_properties(self):
+        """
+        A :class:`PublishData` instance that houses item properties local to
+        the current publish plugin instance. As such, it is expected that this
+        property is only accessed from within a publish plugin. Attempts to
+        access this property outside of a publish plugin will raise an
+        ``AttributeError``.
+
+        This property behaves like the local storage in python's threading
+        module, except here, the data is local to the current publish plugin.
+
+        You can get and set values for this property using standard dictionary
+        notation or via dot notation.
+
+        It is important to consider when to set a value via
+        :meth:`Item.properties`` and when to use :meth:`Item.local_properties`.
+        Setting the values on ``item.properties`` is a way to globally share
+        information between publish plugins. Values set via
+        ``item.local_properties`` will only be applied during the execution of
+        the current plugin (similar to python's ``threading.local`` storage).
+
+        A common scenario to consider is when you have multiple publish plugins
+        acting on the same item. You may, for example, want the ``publish_name``
+        and ``publish_version`` to be shared by each plugin, while setting the
+        remaining properties on each plugin instance since they will be specific
+        to that plugin's output.
+
+        Example::
+
+            # set shared properties on the item (potentially in the collector or
+            # the first publish plugin). these values will be available to all
+            # publish plugins attached to the item.
+            item.properties.publish_name = "Gorilla"
+            item.properties.publish_version = "0003"
+
+            # set specific properties in subclasses of the base file publish (this
+            # class). first publish plugin...
+            item.local_properties.publish_template = "asset_fbx_template"
+            item.local_properties.publish_type = "FBX File"
+
+            # in another publish plugin...
+            item.local_properties.publish_template = "asset_abc_template"
+            item.local_properties.publish_type = "Alembic Cache"
+        """
+        return self._get_local_properties()
+
+    @property
+    def name(self):
+        """The display name of the item."""
+        return self._name
+
+    @name.setter
+    def name(self, new_name):
+        """Sets a new display name for the item with the given string."""
+        self._name = new_name
+
+    @property
+    def parent(self):
+        """The item's parent item."""
+        return self._graph.get_parent(self)
+
+    @property
+    def persistent(self):
```

Comment:
This is a new property for items that should not be removed between collections (like dragged/dropped files). It's a more generic approach than what we did manually in the pre existing code. This also allows clients who build/manage their graph can mark any items as persistent if they wish.